extern/boostd-data/bench/run.go

```diff
@@ -0,0 +1,291 @@
+	bitswapFetchParallelism   int
+	graphsyncFetchCount       int
+	graphsyncFetchParallelism int
+}
+
+func run(ctx context.Context, db BenchDB, opts runOpts) error {
+	log.Infof("Running benchmark for %s", db.Name())
+	log.Infof("Initializing...")
+	if err := db.Init(ctx); err != nil {
+		return err
+	}
+	log.Infof("Initialized")
+
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		if err := db.Cleanup(ctx); err != nil {
+			log.Errorf("cleaning up database: %w", err)
+		}
+	}()
+
+	// Add sample data to the database
+	if err := addPieces(ctx, db, opts.pieceParallelism, opts.pieceCount, opts.blocksPerPiece); err != nil {
+		return err
+	}
+
+	// Run bitswap fetch simulation
+	if err := bitswapFetch(ctx, db, opts.bitswapFetchCount, opts.bitswapFetchParallelism); err != nil {
+		return err
+	}
+
+	// Run graphsync fetch simulation
+	if err := graphsyncFetch(ctx, db, opts.graphsyncFetchCount, opts.graphsyncFetchParallelism); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func loadCmd(createDB func(context.Context, string) (BenchDB, error)) *cli.Command {
+	return &cli.Command{
+		Name:   "load",
+		Before: before,
+		Action: func(cctx *cli.Context) error {
+			ctx := cliutil.ReqContext(cctx)
+			db, err := createDB(ctx, cctx.String("connect-string"))
+			if err != nil {
+				return err
+			}
+
+			opts := runOptsFromCctx(cctx)
+			return addPieces(ctx, db, opts.pieceParallelism, opts.pieceCount, opts.blocksPerPiece)
+		},
+	}
+}
+
+func addPieces(ctx context.Context, db BenchDB, parallelism int, pieceCount int, blocksPerPiece int) error {
+	log.Infof("Adding %d pieces with %d blocks per piece...", pieceCount, blocksPerPiece)
+
+	queue := make(chan struct{}, pieceCount)
+	for i := 0; i < pieceCount; i++ {
+		queue <- struct{}{}
+	}
+	close(queue)
+
+	var totalCreateRecs, totalAddRecs time.Duration
+	addStart := time.Now()
+	var eg errgroup.Group
+	baseCid := testutil.GenerateCid().Bytes()
+	for i := 0; i < parallelism; i++ {
+		eg.Go(func() error {
+			for {
+				select {
+				case <-ctx.Done():
+					return ctx.Err()
+				case _, ok := <-queue:
+					if !ok {
+						return nil
+					}
+
+					// Create block records
+					createRecsStart := time.Now()
+
+					recs := make([]model.Record, 0, blocksPerPiece)
+					for p := 0; p < blocksPerPiece; p++ {
+						c, err := generateRandomCid(baseCid)
+						if err != nil {
+							return err
+						}
+
+						recs = append(recs, model.Record{
+							Cid: c,
+							OffsetSize: model.OffsetSize{
+								Offset: uint64(rand.Intn(sectorSize)),
+								Size:   uint64(rand.Intn(sectorSize)),
+							},
+						})
+					}
+					totalCreateRecs += time.Since(createRecsStart)
+
+					// Add the records to the db
+					addRecsStart := time.Now()
+					pcid := testutil.GenerateCid()
+					err := db.AddIndexRecords(ctx, pcid, recs)
+					if err != nil {
+						return err
+					}
+					totalAddRecs += time.Since(addRecsStart)
```

Comment:
Would also log time+record count here so it's possible to see perf over time, and see how batch size / parallelism / etc. impact individual batch perf

Reply:
We're planning to test with 100s of millions of records so we'll probably need to do something in an aggregate fashion